INTRODUCTION.md

```diff
@@ -705,15 +705,106 @@ which returns one of the following values:
 This method should be called by the application on delivery report error.
 
 
+### Transactional Producer
 
+#### FIXME: misc
+
+To make sure messages time out (in case of connectivity problems, etc) within
+the transaction, the `message.timeout.ms` configuration property must be
+set lower than the `transaction.timeout.ms`.
+If `message.timeout.ms` is not explicitly configured it will be adjusted
+automatically.
+
+
+#### Old producer fencing
+
+If a new transactional producer instance is started with the same
+`transactional.id`, any previous still running producer
+instance will be fenced off at the next produce, commit or abort attempt, by
+raising a fatal error with the error code set to
+`RD_KAFKA_RESP_ERR__FENCED`.
+
+
+
+### Exactly Once Semantics (EOS) and transactions
+
+librdkafka supports Exactly One Semantics (EOS) as defined in [KIP-98](https://cwiki.apache.org/confluence/display/KAFKA/KIP-98+-+Exactly+Once+Delivery+and+Transactional+Messaging).
+For more on the use of transactions, see [Transactions in Apache Kafka](https://www.confluent.io/blog/transactions-apache-kafka/).
+
+
+The transactional consume-process-produce loop roughly boils down to the
+following pseudo-code:
+
+```c
+    /* Producer */
+    rd_kafka_conf_t *pconf = rd_kafka_conf_new();
+    rd_kafka_conf_set(pconf, "bootstrap.servers", "mybroker");
+    rd_kafka_conf_set(pconf, "transactional.id", "my-transactional-id");
+    rd_kafka_t *producer = rd_kafka_new(RD_KAFKA_PRODUCER, pconf);
+
+    rd_kafka_init_transactions(producer);
+
+
+    /* Consumer */
+    rd_kafka_conf_t *cconf = rd_kafka_conf_new();
+    rd_kafka_conf_set(cconf, "bootstrap.servers", "mybroker");
+    rd_kafka_conf_set(cconf, "group.id", "my-group-id");
+    rd_kafka_conf_set(cconf, "enable.auto.commit", "false");
```

Comment:
... this is an unfortunate api flaw. `enable.auto.commit=true` is such a bad default.... this is just one more reason why.